src/main/java/bio/terra/workspace/app/controller/WorkspaceApiController.java

```diff
@@ -78,4 +81,19 @@ private AuthenticatedUserRequest getAuthenticatedInfo() {
         jobService.retrieveJobResult(id, Object.class, userReq);
     return new ResponseEntity<>(jobResultHolder.getResult(), jobResultHolder.getStatusCode());
   }
+
+  @Override
+  public ResponseEntity<DataReferenceList> enumerateReferences(
+      @PathVariable("id") String id,
+      @Valid @RequestParam(value = "offset", required = false, defaultValue = "0") Integer offset,
+      @Valid @RequestParam(value = "limit", required = false, defaultValue = "10") Integer limit,
+      @Valid @RequestParam(value = "filterControlled", required = false, defaultValue = "all")
```

Comment:
Do you expect to have other kinds of filters eventually? This one seems narrow.

Reply:
I was planning to add other filters as separate params in the future. I'm not sure which fields we need to filter on yet, or if there's a cleaner way to filter on multiple settings than just using multiple params.

Reply:
for forward compatibility, how about a generic "filter" parameter, which accepts a JSON key-value map? Or some other form of generic/expandable payload like a query DSL?

Reply:
A key-value map seems like a nice, generic approach and also lets us move the filter parameter to a shared definition. The only drawback seems like we'd be validating in the service itself instead of right at the API, but that should be manageable with documentation.

Reply:
Thinking about this more: a JSON kv map seems like a generally good solution, but I don't have a great sense of all the filtering use cases we'll want to cover in the future, and guessing about those before they show up kind of feels like premature optimization. Given that filtering is explicitly out of scope for AS271, I'm leaning towards removing it from this PR entirely. 
Does that seem reasonable?